presto-main/src/main/java/com/facebook/presto/execution/scheduler/group/DynamicLifespanScheduler.java

```diff
@@ -151,8 +167,19 @@ public void onTaskFailed(int taskId, List<SourceScheduler> sourceSchedulers)
                 for (SourceScheduler sourceScheduler : sourceSchedulers) {
                     sourceScheduler.rewindLifespan(Lifespan.driverGroup(driverGroupId), partitionHandles.get(driverGroupId));
                 }
-                driverGroupQueue.enqueue(driverGroupId);
+                noPreferenceDriverGroups.enqueue(driverGroupId);
+            }
+
+            // when a task fails, all driverGroups that prefer this task/node would be not able to execute
```

Comment:
nit: Let's follow the other comments' pattern and capitalize the first character of the sentence :)